dessia_common/core.py

```diff
@@ -557,13 +557,20 @@ def zip_settings(self):
         Excel file stream generated by calling 'to_xlsx_stream' method.
         JSON file stream generated by calling 'save_to_stream' method.
         """
-        excel_stream = dcf.BinaryFile("excel_export.xlsx")
-        self.to_xlsx_stream(excel_stream)
-
-        json_stream = dcf.JsonFile("json_export.json")
-        self.save_to_stream(json_stream)
+        list_stream = []
+        for export_format in self._export_formats():
+            if export_format.extension != "zip":
+                method_name = export_format.method_name
+                if export_format.text:
```

Comment:
This statements can be simplified. Consider this, instead : 

```py
stream_class = dcf.StringFile if export_format.text else dcf.BinaryFile
stream = stream_class(filename=f"export.{export_format.extension}")
getattr(self, method_name)(stream)
streams.append(stream)
```

Reply:
Yes !
done !